src/object/authenticate_access_auth.cpp

```diff
@@ -1136,12 +1175,691 @@ au_user_revoke_all_privileges (MOP user_mop)
+
+  AU_DISABLE (save);
+
+  db_make_null (&val);
+  db_make_null (&db_auth_object_value);
+  db_make_null (&grantor_value);
+  db_make_null (&grantee_value);
+  db_make_null (&object_of_value);
+  db_make_null (&auth_type_value);
+  db_make_null (&is_grantable_value);
+
+  switch (obj_type)
+    {
+    case DB_OBJECT_CLASS:
+      sprintf (obj_fetch_query, sql_query, "SELECT [c].[class_of] FROM " CT_CLASS_NAME "[c] WHERE [unique_name] = ?");
+      break;
+    case DB_OBJECT_PROCEDURE:
+      sprintf (obj_fetch_query, sql_query, "SELECT [sp] FROM " CT_STORED_PROC_NAME "[sp] WHERE [unique_name] = ?");
+      break;
+    default:
+      error = ER_FAILED;
+      ASSERT_ERROR_AND_SET (error);
+      goto exit;
+    }
+
+  session = db_open_buffer_local (obj_fetch_query);
+  if (session == NULL)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      goto exit;
+    }
+
+  error = db_set_system_generated_statement (session);
+  if (error != NO_ERROR)
+    {
+      goto release;
+    }
+
+  stmt_id = db_compile_statement_local (session);
+  if (stmt_id < 0)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      goto release;
+    }
+
+  /* Prepare DB_VALUEs for host variables */
+  db_make_string (&val, unique_name);
+
+  error = db_push_values (session, 1, &val);
+  if (error != NO_ERROR)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      goto release;
+    }
+
+  error = db_execute_statement_local (session, stmt_id, &result);
+
+  /* The error value is row count if it's not negative value. */
+  if (error == 0)
+    {
+      error = ER_GENERIC_ERROR;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 0);
+      goto release;
+    }
+  else if (error < 0)
+    {
+      ASSERT_ERROR_AND_SET (error);
+      goto release;
+    }
+
+  error = NO_ERROR;
+  while (db_query_next_tuple (result) == DB_CURSOR_SUCCESS)
+    {
+      if (db_query_get_tuple_value (result, 0, &db_auth_object_value) == NO_ERROR)
+	{
+	  if (!DB_IS_NULL (&db_auth_object_value))
+	    {
+	      db_auth_object_mop = db_get_object (&db_auth_object_value);
+	    }
+	  else
+	    {
+	      goto release;
+	    }
+	}
+
+      if (db_query_get_tuple_value (result, 1, &grantor_value) == NO_ERROR)
+	{
+	  if (!DB_IS_NULL (&grantor_value))
+	    {
+	      grantor_mop = db_get_object (&grantor_value);
+	    }
+	  else
+	    {
+	      goto release;
+	    }
+	}
+
+      if (db_query_get_tuple_value (result, 2, &grantee_value) == NO_ERROR)
+	{
+	  if (!DB_IS_NULL (&grantee_value))
+	    {
+	      grantee_mop = db_get_object (&grantee_value);
+	    }
+	  else
+	    {
+	      goto release;
+	    }
+	}
+
+      if (db_query_get_tuple_value (result, 3, &object_of_value) == NO_ERROR)
+	{
+	  if (!DB_IS_NULL (&object_of_value))
+	    {
+	      object_of_mop = db_get_object (&object_of_value);
+	    }
+	  else
+	    {
+	      goto release;
+	    }
+	}
+
+      if (db_query_get_tuple_value (result, 4, &auth_type_value) == NO_ERROR)
+	{
+	  auth_type_char = NULL;
+
+	  if (!DB_IS_NULL (&auth_type_value))
```

Comment:
```suggestion
	  if (DB_IS_NULL (&auth_type_value))
```
다음과 같이 수정 후 에러 처리를 하고, 현재 else문에 있는 블럭을 밖으로 빼내는 건 어떨까요?

Reply:
코드 리뷰 감사합니다.
리뷰 내용을 참고하여, 아래 함수에도 동일하게 반영하여 수정 했습니다.
1. au_object_revoke_all_privileges()
2. au_user_revoke_all_privileges()
3. update_authorization_for_new_owner()
4. update_auth_for_new_owner()